src/System.Security.Cryptography.DeriveBytes/src/System/Security/Cryptography/Rfc2898DeriveBytes.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text;
+using System.Diagnostics;
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public class Rfc2898DeriveBytes : DeriveBytes
+    {
+        public Rfc2898DeriveBytes(byte[] password, byte[] salt, int iterations)
+        {
+            Salt = salt;
+            IterationCount = iterations;
+            if (password == null)
+                throw new NullReferenceException();  // This "should" be ArgumentNullException but for compat, we throw NullReferenceException.
+            _password = password.CloneByteArray();
+            _hmacSha1 = new HMACSHA1(_password);
+
+            // We "should" call Initialize() here but we've already called it twice indirectly through setting the Salt and IterationCount properties.
+            return;
+        }
+
+        public Rfc2898DeriveBytes(String password, byte[] salt)
+             : this(password, salt, 1000)
+        {
+        }
+
+        public Rfc2898DeriveBytes(String password, byte[] salt, int iterations)
+            : this(new UTF8Encoding(false).GetBytes(password), salt, iterations)
+        {
+        }
+
+        public Rfc2898DeriveBytes(String password, int saltSize)
+            : this(password, saltSize, 1000)
+        {
+        }
+
+        public Rfc2898DeriveBytes(String password, int saltSize, int iterations)
+        {
+            if (saltSize < 0)
+                throw new ArgumentOutOfRangeException("saltSize", SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            Salt = Helpers.GenerateRandom(saltSize);
+            IterationCount = iterations;
+            _password = new UTF8Encoding(false).GetBytes(password);
+            _hmacSha1 = new HMACSHA1(_password);
+
+            // We "should" call Initialize() here but we've already called it twice indirectly through setting the Salt and IterationCount properties.
```

Comment:
Instead of setting Salt and IterationCount then, how about setting _salt and _iterations and then calling Initialize, so that we don't initialize multiple times unnecessarily?